File: utils/pose_logger_NOT_TESTED.py
import rclpy
from geometry_msgs.msg import PoseStamped
from sys import argv

def pose_callback(msg):
    with open(argv[1], 'a') as file:
        file.write(f"{msg.header.stamp.sec}.{msg.header.stamp.nanosec},{msg.pose.position.x},{msg.pose.position.y},{msg.pose.position.z}\n")

def main():
    rclpy.init(args=None)
    node = rclpy.create_node('pose_logger')
    pose_sub = node.create_subscription(PoseStamped, argv[2], pose_callback, 10)

    while rclpy.ok():
        rclpy.spin_once(node)

if __name__ == '__main__':
    if len(argv) != 3:
        print("Usage: python pose_logger.py <output_file> <pose_topic>")
    else:
        main()

File: utils/pose_logger_NOT_TESTED.cpp
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/pose_stamped.hpp"
#include <fstream>
#include <string>

class PoseLogger : public rclcpp::Node
{
public:
    explicit PoseLogger(const std::string& output_file, const std::string& pose_topic)
    : Node("pose_logger"), output_file_(output_file), pose_topic_(pose_topic)
    {
        pose_subscriber_ = this->create_subscription<geometry_msgs::msg::PoseStamped>(
            pose_topic_, 10, std::bind(&PoseLogger::pose_callback, this, std::placeholders::_1));
    }

private:
    void pose_callback(const geometry_msgs::msg::PoseStamped::SharedPtr msg)
    {
        std::ofstream file(output_file_, std::ios_base::app);
        if (file.is_open()) {
            file << msg->header.stamp.sec << "." << msg->header.stamp.nanosec << ","
                 << msg->pose.position.x << "," << msg->pose.position.y << "," << msg->pose.position.z << "\n";
        }
    }

    rclcpp::Subscription<geometry_msgs::msg::PoseStamped>::SharedPtr pose_subscriber_;
    std::string output_file_;
    std::string pose_topic_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    if (argc != 3) {
        std::cerr << "Usage: " << argv[0] << " <output_file> <pose_topic>" << std::endl;
        return 1;
    }
    auto node = std::make_shared<PoseLogger>(std::string(argv[1]), std::string(argv[2]));
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
